habitat-lab/habitat/tasks/rearrange/utils.py

```diff
@@ -785,3 +606,197 @@ def get_camera_lookat_relative_to_vertial_line(
     # Get angle between location and the vector
     angle = get_camera_object_angle(cam_T, vertical_dir, local_vertical_dir)
     return angle
+
+
+def embodied_unoccluded_navmesh_snap(
```

Comment:
My primary concern about this change is the behavior is fundamentally different from the previous `_get_robot_spawns`. If robot spawns failed, it would still return _some_ starting configuration for the robot. However, if this fails, it will return None, and presumably the robot will not be moved. I think it's dangerous not to have this fall back of placing the robot at least somewhere reasonable, even if it's in collision. Not moving the robot at all can cause big problems in reward calculation that are hard to catch.